### DISCUSSION TITLE
Granting multiple roles to a single user

### DISCUSSION THREAD

**Discussion by hieu-cc** [Snowflake-Labs/terraform-provider-snowflake#2200] [Q&A]
Is there a resource to allow granting multiple roles to a user?

Something like [snowflake_role_grants](https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs/resources/role_grants) but for one user?

Currently, if I want to do this with `snowflake_role_grants`, then I need to create multiple resources of this type for my user. Which makes the plan more difficult to read if we have many of them:
```
resource "snowflake_role_grants" "role1" {
  role_name = "role1"
  users = ["user1"]
}

resource "snowflake_role_grants" "role2" {
  role_name = "role2"
  users = ["user1"]
}

resource "snowflake_role_grants" "role3" {
  role_name = "role3"
  users = ["user1"]
}
```

It would be nice to have a resource designated for this. For example:
```
resource "snowflake_user_to_roles_grants" "grants" {
  user = "user1"
  roles = [
    "role1", "role2", "role3",
  ]
}
```

**Comment by bert-ciandt** [comment]
What about using a for_each, like this?

resource "snowflake_role_grants" "grants" {
  for_each = toset(["role1","role2","role3"])
  role_name = each.value
  user = ["user1"]
}

**Comment by hieu-cc** [comment]
I've added some loops and used the transpose() function in Terraform to achieve this.
